tests/test_ci_app_host_home_isolation.py

```
@@ -0,0 +1,97 @@
+)
+
+
+def require(text: str, needle: str, context: str) -> None:
+    if needle not in text:
+        raise SystemExit(f"FAIL: {context} is missing {needle!r}")
+
+
+def require_step(job_name: str, step_name: str) -> dict:
+    jobs = WORKFLOW.get("jobs")
+    if not isinstance(jobs, dict):
+        raise SystemExit("FAIL: workflow jobs must be a mapping")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)